### PR TITLE
feat(CORE/Config): Added Pet experience rate to config

### PR DESCRIPTION
**Pull request by peti446**
## CHANGES PROPOSED:
- Added Rate.XP.Pet to allow to change the rate at which pets gain experience
- Added Rate.Pet.LevelXP to change the required experience to level up the pet

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3687

## TESTS PERFORMED:
- Captured pet, with default values, checked that it levels up normally
- Changed the rating to a big value like 900 killed enemies and confirm pet leveled up instantly multiple levels 

## HOW TO TEST THE CHANGES:

1. Tame a new pet (/cast tame beast)
2. Kill some enemies and verify pet levels up normally
3. Change rating to a big value
4. .reload config
5. Kill an enemy and confirm pet leveled up very fast
PS: In case needed required experience and current experience for the pet can be obtained by writing in chat: /dump GetPetExperience()

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -2392,14 +2392,17 @@ Rate.Drop.Item.ReferencedAmount = 1
...
 #        Description: Experience rates (outside battleground)
 #        Default:     1 - (Rate.XP.Kill)
 #                     1 - (Rate.XP.Quest)
 #                     1 - (Rate.XP.Explore)
+#                     1	- (Rate.XP.Pet)
 
 Rate.XP.Kill    = 1
 Rate.XP.Quest   = 1
 Rate.XP.Explore = 1
+Rate.XP.Pet	 	= 1
```
Comment (Kitzunu) [inline src/server/worldserver/worldserver.conf.dist:2405]: spacing

Comment (FrancescoBorzi) [inline src/server/worldserver/worldserver.conf.dist:2405]: indentation

**Review by FrancescoBorzi — APPROVED** [review]

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -2516,6 +2519,14 @@ Rate.MoveSpeed = 1
 
 Rate.InstanceResetTime = 1
 
+#
+#    Rate.Pet.LevelXP
+#        Description: Modifies the amount of experience required to leveling up a pet.
```
Comment (BarbzYHOOL) [inline src/server/worldserver/worldserver.conf.dist:2524]:
```suggestion
#        Description: Modifies the amount of experience required to level up a pet.
```

File: src/server/worldserver/worldserver.conf.dist [hunk 3]
```diff
@@ -2516,6 +2519,14 @@ Rate.MoveSpeed = 1
 
 Rate.InstanceResetTime = 1
 
+#
+#    Rate.Pet.LevelXP
```
Comment (BarbzYHOOL) [inline src/server/worldserver/worldserver.conf.dist:2523]: hmm why not put in Rate.XP.Pet.PlayerPercentage instead (so all in same code block)
Then you'd rename Rate.XP.Pet to Rate.XP.Pet.Raw

Maybe i'm wrong, just suggesting

File: src/server/worldserver/worldserver.conf.dist (hunk 2 above)
Comment (BarbzYHOOL) [reply src/server/worldserver/worldserver.conf.dist:2524]: I think think it's the experience received by the player that is shared to the pet right? then need to improve description if so and mention the player/character's exp

**Comment by BarbzYHOOL** [comment]
@icenet test this, NOW!!!!!!!

**Comment by IceNet** [comment]
> @IceNet test this, NOW!!!!!!!

Sir, yes, Sir!
Works fine!

**Comment by Helias** [comment]
very nice @IceNet, thank you a lot!

**Comment by FrancescoBorzi** [comment]
> P.S. = the build complains about codefactor, try to fix pls

that's ok, we can ignore it as it complains about existing issues (not introduced by this PR)

**Pull request merged**